sympy/physics/mechanics/particle.py

```diff
@@ -39,56 +41,58 @@ class Particle:
     >>> # Or you could change these later
     >>> pa.mass = m
     >>> pa.point = po
+    >>> # If preferred you can also attach a frame to a Particle
+    >>> pa.add_frame()
+    >>> pa.frame
+    pa_frame
 
     """
+    point = _Body.masscenter
 
-    def __init__(self, name, point, mass):
-        if not isinstance(name, str):
-            raise TypeError('Supply a valid name.')
-        self._name = name
-        self.mass = mass
-        self.point = point
-        self.potential_energy = 0
-
-    def __str__(self):
-        return self._name
-
-    def __repr__(self):
-        return self.__str__()
+    def __init__(self, name, point=None, mass=None):
+        super().__init__(name, point, mass)
+        self._frame = None
 
     @property
-    def mass(self):
-        """Mass of the particle."""
-        return self._mass
-
-    @mass.setter
-    def mass(self, value):
-        self._mass = sympify(value)
-
-    @property
-    def point(self):
-        """Point of the particle."""
-        return self._point
-
-    @point.setter
-    def point(self, p):
-        if not isinstance(p, Point):
-            raise TypeError("Particle point attribute must be a Point object.")
-        self._point = p
+    def frame(self):
+        """The optional ReferenceFrame fixed to the Particle."""
+        if self._frame is None:
+            raise AttributeError("Particle has no attached frame.")
+        return self._frame
+
+    @frame.setter
+    def frame(self, frame):
+        if frame is None:
+            self._frame = None
+            return
+        elif isinstance(frame, ReferenceFrame):
+            self._frame = frame
+        else:
+            self._frame = ReferenceFrame(frame)
+        self.point.set_vel(frame, 0)
+
+    def add_frame(self, frame=None):
+        """Method to attach a frame to the particle."""
+        if frame is None:
+            if self._frame is not None:
+                return
+            frame = ReferenceFrame(f'{self.name}_frame')
+        self.frame = frame
```

Comment:
Would it be nice for this method to return the newly-added frame?

Reply:
Yeah, could do. Note that this method does not always create a new frame (if no frame is supplied and there is already a frame).